integration/query_logical_test.go

```diff
@@ -0,0 +1,298 @@
+					bson.D{{"value", bson.D{{"$lt", 0}}}},
+					bson.D{{"value", bson.D{{"$lt", 42}}}},
+				},
+			}},
+			expectedIDs: []any{
+				"double-negative-infinity", "double-negative-zero",
+				"double-smallest", "double-zero",
+				"int32-min", "int32-zero", "int64-min", "int64-zero",
+			},
+		},
+		"BadInput": {
+			filter: bson.D{{"$or", nil}},
+			err: mongo.CommandError{
+				Code:    2,
+				Message: "$or must be an array",
+				Name:    "BadValue",
+			},
+		},
+		"BadExpressionValue": {
+			filter: bson.D{{
+				"$or",
+				bson.A{
+					bson.D{{"value", bson.D{{"$gt", 0}}}},
+					nil,
+				},
+			}},
+			err: mongo.CommandError{
+				Code:    2,
+				Message: "$or/$and/$nor entries need to be full objects",
+				Name:    "BadValue",
+			},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var actual []bson.D
+			cursor, err := collection.Find(ctx, tc.filter)
+			if tc.err.Code != 0 {
+				require.Nil(t, tc.expectedIDs)
+				assertEqualError(t, tc.err, err)
+				return
+			}
+			require.NoError(t, err)
+			err = cursor.All(ctx, &actual)
+			require.NoError(t, err)
+			assert.Equal(t, tc.expectedIDs, collectIDs(t, actual))
+		})
+	}
+}
+
+func TestQueryLogicalNor(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setupWithOpts(t, &setupOpts{providers: []shareddata.Provider{shareddata.Scalars}})
+
+	for name, tc := range map[string]struct {
+		filter      bson.D
+		expectedIDs []any
+		err         mongo.CommandError
+	}{
+		"Nor": {
+			filter: bson.D{{
+				"$nor",
+				bson.A{
+					bson.D{{"value", bson.D{{"$gt", 0}}}},
+					bson.D{{"value", bson.D{{"$gt", 42}}}},
+				},
+			}},
+			expectedIDs: []any{
+				"binary", "binary-empty", "bool-false", "bool-true",
+				"datetime", "datetime-epoch", "datetime-year-max", "datetime-year-min",
+				"double-nan", "double-negative-infinity", "double-negative-zero", "double-zero",
+				"int32-min", "int32-zero", "int64-min", "int64-zero",
+				"null", "objectid", "objectid-empty",
+				"regex", "regex-empty", "string", "string-double", "string-empty", "string-whole",
+				"timestamp", "timestamp-i",
+			},
+		},
+		"BadInput": {
+			filter: bson.D{{"$nor", nil}},
+			err: mongo.CommandError{
+				Code:    2,
+				Message: "$nor must be an array",
+				Name:    "BadValue",
+			},
+		},
+		"BadExpressionValue": {
+			filter: bson.D{{
+				"$nor",
+				bson.A{
+					bson.D{{"value", bson.D{{"$gt", 0}}}},
+					nil,
+				},
+			}},
+			err: mongo.CommandError{
+				Code:    2,
+				Message: "$or/$and/$nor entries need to be full objects",
+				Name:    "BadValue",
+			},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var actual []bson.D
+			cursor, err := collection.Find(ctx, tc.filter, options.Find().SetSort(bson.D{{"_id", 1}}))
+			if tc.err.Code != 0 {
+				require.Nil(t, tc.expectedIDs)
+				assertEqualError(t, tc.err, err)
+				return
+			}
+			require.NoError(t, err)
+			err = cursor.All(ctx, &actual)
```

Comment:
Notice how `actual` is defined way above, no blank line, additional `require.NoError(t, err)`, etc

It seems like we continue to copy&paste two different styles instead of reunifiying them

Reply:
Fixed